tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc

```diff
@@ -132,6 +149,44 @@ Status PollFileSystemForServable(
     LOG(WARNING) << "No versions of servable " << servable.servable_name()
                  << " found under base path " << servable.base_path();
   }
+}
+
+// Like PollFileSystemForConfig(), but for servable
+Status PollFileSystemForServable(
+    const FileSystemStoragePathSourceConfig::ServableToMonitor& servable,
+    std::vector<ServableData<StoragePath>>* versions) {
+  // First, determine whether the base path exists. This check guarantees that
+  // we don't emit an empty aspired-versions list for a non-existent (or
+  // transiently unavailable) base-path. (On some platforms, GetChildren()
+  // returns an empty list instead of erring if the base path isn't found.)
+  if (!Env::Default()->FileExists(servable.base_path())) {
+    return errors::InvalidArgument("Could not find base path ",
+                                   servable.base_path(), " for servable ",
+                                   servable.servable_name());
+  }
+
+  // Retrieve a list of base-path children from the file system.
+  std::vector<string> children;
+  TF_RETURN_IF_ERROR(
+      Env::Default()->GetChildren(servable.base_path(), &children));
+
+  // GetChildren() returns all descendants instead for cloud storage like GCS.
+  // In such case we should filter out all non-direct descendants.
+  std::set<string> real_children;
+  for (int i = 0; i < children.size(); ++i) {
+    const string& child = children[i];
+    real_children.insert(child.substr(0, child.find_first_of('/')));
+  }
+  children.clear();
+  children.insert(children.begin(), real_children.begin(), real_children.end());
+
+  LOG(INFO) << "Version policy of servable:" << servable.version_policy();
+
+  if (servable.version_policy() == FileSystemStoragePathSourceConfig::ALL_VERSION) {
```

Comment:
Please use a switch statement. That way if, in the future, we add another case, the compiler will force us to examine this code and consider how to handle the new case.